### PR TITLE
chore: pin GitHub actions with their SHA-1 instead of their version number

### PR DESCRIPTION
**Pull request by lemeurherve**
Ref: https://github.com/jenkins-infra/helpdesk/issues/3402

### CODE REVIEW

File: .github/workflows/jenkins-security-scan.yaml [hunk 1]
```diff
@@ -31,7 +27,7 @@ jobs:
           repository: jenkins-infra/jenkins-codeql
           path: jenkins-security-scan-rules
       - name: Install CodeQL CLI
-        uses: jenkins-infra/fetch-codeql-action@v1
+        uses: jenkins-infra/fetch-codeql-action@9c7fa19065a1a63d60f1f4337ec1e03955842f2a # v1
```
Comment (daniel-beck) [inline .github/workflows/jenkins-security-scan.yaml:30]: Same org, same maintainers, this isn't necessary IMO.

File: .github/workflows/jenkins-security-scan.yaml [hunk 2]
```diff
@@ -78,7 +74,7 @@ jobs:
             jenkins-security-scan.sarif
           name: Jenkins Security Scan SARIF
       - name: Upload Scan Result
-        uses: github/codeql-action/upload-sarif@v2
+        uses: github/codeql-action/upload-sarif@17573ee1cc1b9d061760f3a006fc4aac4f944fd5 # v2
```
Comment (daniel-beck) [inline .github/workflows/jenkins-security-scan.yaml:77]: Official GH action, seems unnecessary.

**Review by daniel-beck — COMMENTED** [review]
I won't apply this for the reasons given below, but thanks for taking the time to submit this for consideration :)

**Comment by lemeurherve** [comment]
You're right, these actions shouldn't have been concerned by the SHA-1 pinning.

See my comment agreeing with you here: https://github.com/jenkins-infra/helpdesk/issues/3402#issuecomment-1443742488

**Pull request closed without merging**